samcli/commands/init/__init__.py

```diff
@@ -149,15 +149,13 @@ def do_cli(
     # check for required parameters
     if location or (name and runtime and dependency_manager and app_template):
         # need to turn app_template into a location before we generate
+        default_context = None
         if app_template:
             templates = InitTemplates(no_interactive, auto_clone)
             location = templates.location_from_app_template(runtime, dependency_manager, app_template)
             no_input = True
             default_context = {"project_name": name, "runtime": runtime}
```

Comment:
why is default_context only initialized with the name/runtime parameters in the app_template flow? is there no expetations to be provided those parameters in location flow? I suppose I can understand, but is that a documented assumption? 

https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-init.html only says that "This parameter is required if --no-interactive is specified and --runtime, --name, and --app-template are not provided.", but that doesn't necessarily imply that the reverse is true.

Should we just use this as the default in all cases? (assuming there is a name && runtime)

Reply:
As per my understanding currently other parameters like runtime, name are use to determine template and then passed in cookie cutter template. Adding them in location flow means I am assuming customer is going to use these parameters in cookie cutter template.

Reply:
for sure - "location" can be any github repo or directory, and they could have parameterized runtime and project_name or others.

might be a safe call

Reply:
I agree with @nikp here. I don't think `{"project_name": name, "runtime": runtime}` is specific to app_template and should be moved up to be included with `--location`. This gives the ability to other authors to have access to these values.

Reply:
User will have access to {"project_name": name, "runtime": runtime} by setting this as part of extra_context. I am assuming SAM init with location will only use extra_context no other parameter. If it's not true I am happy to make it as default parameter.

Reply:
But they can also pass `--name` and `--runtime` into the command. Why would we want customers to be duplicating these values? How will customers know they need to do that vs just pass the option to the command?

Reply:
In case of location if customer passes name and runtime we assume its cookie cutter template variable is it right understanding?

Reply:
I am happy to update if this is the expectation.

Reply:
I think it's reasonable to map the `--name` param to the context `project_name` and the `--runtime` param to the `runtime` context.

Reply:
@gupta-n Yes. We want to start passing this information to the templates that are given through `--location`. This should simplify some of this code as well.